steps/index.lab.md

```diff
@@ -54,93 +53,154 @@ $ git clone https://github.com/firebase/codelab-friendlychat-android
+1. Start your Android device or emulator.
+2. In Android Studio click **Run** ( <img src="img/execute.png" alt="execute"  width="20.00" />) in the toolbar.
 
 The app should launch on your device. At this point, you should see an empty message list, and sending and receiving messages will not work. In the next section, you authenticate users so they can use Friendly Chat.
 
-
 ## Enable Authentication
 Duration: 05:00
 
+This app will use Firebase Realtime Database to store all chat messages. Before we add data, we should make sure that the app is secure and that only authenticated users can post messages. In this step we will enable Firebase Authentication and configure Realtime Database Security Rules.
 
-Let's require a user to sign in before reading or posting any Friendly Chat messages.
+### Configure Firebase Authentication
 
-#### **Firebase Realtime Database Rules**
+Before your application can access the Firebase Authentication APIs on behalf of your users, you will have to enable it 
 
-Access to your Firebase Database is configured by a set of rules written in a JSON configuration language.
+1. Navigate to the  [Firebase console](http://console.firebase.google.com) and select your project
+2. Select **Authentication**
+3. Select the **Sign In Method** tab
+4. Toggle the **Google** switch to enabled (blue)
+5. Set a support email.
+6. Press **Save** on the resulting dialog
+
+If you get errors later in this codelab with the message "CONFIGURATION_NOT_FOUND", come back to this step and double check your work.
 
-Go to your project in the Firebase console and select **Database**. Select the **Realtime Database** option (not Cloud Firestore). If prompted for security rules, with choices to start in either **test mode** or **locked mode**, choose **locked mode**. Once the default rules are established, select the **Rules** tab and update the rules configuration with the following:
+### Configure Realtime Database
+
+As mentioned, this app will store chat messages in Firebase Realtime Database. In this step we will create a database and configure the security via a JSON configuration language called Security Rules.
+
+1. Go to your project in the Firebase console and select **Realtime Database** from the left navigation.
+2. Click **Create Database** create a new Realtime Database instance and then select the `us-central1` region and click **Next**.
+3. When prompted about security rules, choose **locked mode** and click **Enable**. 
+
+Once the database has been created, select the **Rules** tab and update the rules configuration with the following:
 
 ```
 {
   "rules": {
-    ".read": "auth != null",
-    ".write": "auth != null"
+    "messages": {
+      "$message": {
+        ".read": "auth != null",
+        ".write": "auth != null"
```

Comment:
Is there some way of making this more secure from the beginning. All users should be able to read all messages to the `read` rule is fine but maybe we can update the `write` one to be a bit more specific?

Reply:
I would love to but it turns out the data model doesn't currently store the UID or anything so ... there's not a good way to ensure that users can only write good messages.

Changing the data model is out of scope because we want it to be compatible across iOS and Web.